src/modules/job-list/job_state.c

```diff
@@ -181,18 +181,27 @@ static void job_insert_list (struct job_state_ctx *jsctx,
         if (!(job->list_handle = zlistx_insert (jsctx->pending,
                                                 job,
                                                 search_direction (job))))
-            flux_log (jsctx->h, LOG_ERR, "%s: zlistx_insert", __FUNCTION__);
+            flux_log (jsctx->h,
+                      LOG_ERR,
+                      "%s: zlistx_insert: out of memory",
```

Comment:
Well...  this function has three possible failures that are all ENOMEM.  The error messages are going to be lke
```
job-list: job_insert_list: zlistx_insert: out of memory
```
Not much context.  Also, distinguishing the three separate ENOMEM cases with unique log messages is also probably unhelpful.

Suggestion: change  `job_insert_list()` to return success/failure and set errno = ENOMEM on failure.  Then handle the logging in the callers.   One caller is `job_change_list()` which could supply more context like:
```
job-list: error moving job to new list on state transition to <state>: <strerror>
```
The other caller is `depthfirst_map_one()`, maybe:
```
job-list: error moving job to new list during restart from kvs:  <strerror>
```

Reply:
good idea, i did a similar "collapse" in another function.  I'll do it here too.